### ISSUE TITLE
Cannot get Porcupine wakeword detection working with Raspberry-pi

### ISSUE THREAD

**Issue by butteryak** [MycroftAI/mycroft-core#2720]
Trying to use Porcupine for wake word detection, and not having any success:  As there is no way to create a custom wakeword as optimizer is no longer part of the porcupine package, I do see there is a way with the picovoice console tool to create custom wake word, but the free version only allows creation of wake for for x86 not ARM based machines....so trying to work with the existing models that are included with Procupine package.  I have set mycroft.conf as this:

```
"listener": {
  "wake_word": "blueberry"
},

"hotwords": {
  "blueberry": {
    "module": "porcupine",
    "porcupine_path": "~/.mycroft/Porcupine",
    "keyword_file_path": "~/.mycroft/Porcupine/resources/keyword_files/raspberry-pi/blueberry_raspberry-pi.ppn"
    }
}
```

Porcupine wake word appears to be attempint to load, but having this error:

```
2020-10-13 18:03:22.296 | INFO     |  6577 | mycroft.client.speech.listener:create_wake_word_recognizer:328 | Creating wake word engine
2020-10-13 18:03:22.299 | INFO     |  6577 | mycroft.client.speech.listener:create_wake_word_recognizer:351 | Using hotword entry for blueberry
2020-10-13 18:03:22.302 | WARNING  |  6577 | mycroft.client.speech.listener:create_wake_word_recognizer:353 | Phonemes are missing falling back to listeners configuration
2020-10-13 18:03:22.305 | WARNING  |  6577 | mycroft.client.speech.listener:create_wake_word_recognizer:357 | Threshold is missing falling back to listeners configuration
2020-10-13 18:03:22.310 | INFO     |  6577 | mycroft.client.speech.hotword_factory:load_module:403 | Loading "blueberry" wake word via porcupine
2020-10-13 18:03:22.315 | INFO     |  6577 | mycroft.client.speech.hotword_factory:__init__:331 | Adding /home/pi/.mycroft/Porcupine/binding/python to Python path
2020-10-13 18:03:22.338 | INFO     |  6577 | mycroft.client.speech.hotword_factory:__init__:356 | Loading Porcupine using library path /home/pi/.mycroft/Porcupine/lib/linux/armv7l/libpv_porcupine.so and keyword paths ['/home/pi/.mycroft/Porcupine/resources/keyword_files/raspberry-pi/blueberry_raspberry-pi.ppn']
2020-10-13 18:03:22.341 | ERROR    |  6577 | mycroft.client.speech.hotword_factory:initialize:423 | Could not create hotword. Falling back to default.
Traceback (most recent call last):
  File "/home/pi/mycroft-core/mycroft/client/speech/hotword_factory.py", line 411, in initialize
    instance = clazz(hotword, config, lang=lang)
  File "/home/pi/mycroft-core/mycroft/client/speech/hotword_factory.py", line 361, in __init__
    sensitivities=sensitivities)
TypeError: __init__() got an unexpected keyword argument 'model_file_path'
2020-10-13 18:03:22.345 | INFO     |  6577 | mycroft.client.speech.hotword_factory:load_module:403 | Loading "blueberry" wake word via pocketsphinx
2020-10-13 18:03:22.475 | INFO     |  6577 | mycroft.client.speech.listener:create_wakeup_recognizer:365 | creating stand up word engine
2020-10-13 18:03:22.478 | INFO     |  6577 | mycroft.client.speech.hotword_factory:load_module:403 | Loading "wake up" wake word via pocketsphinx
2020-10-13 18:03:22.602 | INFO     |  6577 | __main__:on_ready:175 | Speech client is ready.
```

I saw the Phenome and Threshold warnings, so tried to set those in mycroft.conf, but it did not make any difference, I got exactly the same errors. 

Interestingly enough, I ask mycroft what the listener is and it tells me it's "Porcupine"   although any sound seems to trigger it, and it is not recogizing speech commands after trigger. 

I"m noting that the library that is loaded is "Loading Porcupine using library path /home/pi/.mycroft/Porcupine/lib/linux/armv7l/libpv_porcupine.so "   which could be an issue, as this is raspberry-pi and the Procupine package has a specific library for raspberry-pi.    I tried adjusting the code in "hotword_factory.py" to use the raspberry pi library.   

```
 library_path = join(
            porcupine_path, 'lib/raspberry-pi/arm11/libpv_porcupine.so')
        model_file_path = join(
            porcupine_path, 'lib/common/porcupine_params.pv')

```

but I get the same error.   So, how to go about getting this setup on raspberry-pi using the included raspberry-pi wakewords?

_Originally posted by @butteryak in https://github.com/MycroftAI/mycroft-core/pull/2139#issuecomment-707893646_

**Comment by forslund** [comment]
There are a bunch of changes in the porcupine repo since the hotword engine was added which will require some fixes.

I did some additional testing of this and with the latest version on x86_64 (which previously worked) constantly detects wakewords even after fixing the basic issues with missmatched arguments.

**Comment by butteryak** [comment]
Thanks much for checking++   I thought I had set it up correctly, so figured there might be bigger issues. 

If there is any troubleshooting you might need some help with, let me know.   

In the meantime, what version of Porcupine was working?  I may try it out to see if I can make it work.

**Comment by forslund** [comment]
Should mention that it was x86 that was working and not rpi (don't think rpi worked back then either). It was likely v 1.5

I've gotten it running on x86 again with the blueberry hotword. WIP branch [here](https://github.com/forslund/mycroft-core/tree/bugfix/latest-porcupine).

Apart from the parameter names when creating the Porcupine object a small change was needed to make it not always detect the wakeword when checking audio.

Gonna see if I can get it to detect rpi properly or switch to the official python package...

**Comment by forslund** [comment]
Ok I switched to using the library-finder in the porcupine bindings and that finds the library correctly on x86_64. It should technically work on the Pi as well. But I'll need to test (and that will likely not be until tomorrow...

**Comment by forslund** [comment]
Tested it on the Pi and it worked so I've drafted a PR

**Comment by butteryak** [comment]
Tested on raspberry pi.   I just swapped out the "class PorcupineHotWord(HotWordEngine):"  function in the official release on my installation with the function from your dev release and followed your directions from #2722  and it works great.   Still, I'll be stuck with the stock Porcupine wake words, as we can only create free ones for x86, but I get more reliability out of the porcupine module, so I'm good with that for now.  

Thanks much for getting on this so quickly. It's much appreciated, I'm happy with my Porcupine functionality++

I'm assuming my hotword_factory.py "hack"  will get overwritten next time an update happens, but it's an easy fix until this change makes it into the official update.  

Last quick question,   Does the module support setting sensitivity params in mycroft.conf?

thanks again++ works beautifully
